### PR TITLE
chore(*) deprecate cassandra 2.x

### PR DESCRIPTION
**Pull request by kikito**

### CODE REVIEW

File: kong/constants.lua [hunk 1]
```diff
@@ -154,8 +154,8 @@ local constants = {
       -- also accepts a DEPRECATED key, i.e. DEPRECATED = "9.4"
     },
     CASSANDRA = {
-      MIN = "2.2",
-      -- also accepts a DEPRECATED key
+      MIN = "3.0",
+      DEPRECATED = "2.0",
```
[the remark this answers is not in the document]
Comment (kikito) [reply kong/constants.lua]: Grepped the source code and found that Kong doesn't use these constants. Even went further and removed the whole `constant.DATABASE` part - all tests run green. So I have modified the comments to indicate that the constants are only there as a reference.

**Comment by thibaultcha** [comment]
Just curious: why are we deprecating support for 2.2? (Which is supported until the 4.0 release)

File: kong/constants.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (thibaultcha) [reply kong/constants.lua]: Hmm we removed detection for min and deprecated DB versions with the new DAO it seems... It was first added in a9f5af1a53a701bd74c9ed6683390c3f1b43fc6b, and then removed when we dropped the old DAO (34c41ff13a8db8fe365cd73603cdafbbfbd60111). It was an essential check to perform back in the day, we should really add support for these checks back (including deprecation warnings)...

[the remark this answers is not in the document]
Comment (kikito) [reply kong/constants.lua]: @thibaultcha that is out of the scope of this PR, but I have created a new card to track it.

**Comment by kikito** [comment]
@thibaultcha This is not an immediate change, it is targeting Kong 2.2.

**Comment by thibaultcha** [comment]
@kikito I realized that, yes. So what's the driver behind deprecating it then?

File: kong/constants.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (thibaultcha) [reply kong/constants.lua]: Yes of course.

**Comment by hutchic** [comment]
@thibaultcha metrics from splunk indicate 2.2 isn't widely used is what I believe activated this change

**Pull request merged**